src/core/algorithms/statistics/data_stats.cpp

```diff
@@ -505,13 +487,119 @@ Statistic DataStats::GetMedianAD(size_t index) const {
     return Statistic(median_ad, &double_type, false);
 }
 
-Statistic DataStats::GetNumNulls(size_t index) const {
-    if (all_stats_[index].num_nulls.HasValue()) return all_stats_[index].num_nulls;
+Statistic DataStats::GetVocab(size_t index) const {
+    if (all_stats_[index].vocab.HasValue()) return all_stats_[index].vocab;
+    mo::TypedColumnData const& col = col_data_[index];
+    if (col.GetTypeId() != +mo::TypeId::kString) return {};
+
+    mo::StringType string_type;
+    std::string string_data;
+    std::set<char> vocab;
+
+    for (size_t i = 0; i < col.GetNumRows(); i++) {
+        string_data = col.GetDataAsString(i);
```

Comment:
This will count the characters from the string representation of `NullType` (`"NULL"`) as characters in the column. There should be a check with `IsNullOrEmpty`, and then `TypedColumnData::GetValue` + `Type::GetValue<std::string>` can be used to avoid excessive copying

```suggestion
        if (col.IsNullOrEmpty(i)) continue;
        auto const& string_data = mo::Type::GetValue<std::string>(col.GetValue(i));
```